File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/contacts/ui/ContactsPresenterModule.java
package com.fintechplatform.ui.profile.contacts.ui;

import com.fintechplatform.api.profile.api.ProfileAPI;
import com.fintechplatform.ui.models.DataAccount;
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public class ContactsPresenterModule {
    private ContactsContract.View view;
    private DataAccount configuration;

    public ContactsPresenterModule(ContactsContract.View view, DataAccount configuration) {
        this.view = view;
        this.configuration = configuration;
    }

    @Provides
    @Singleton
    ContactsContract.Presenter providesContactsPresenter(ProfileAPI api, UsersPersistanceDB usersPersistanceDB) {
        return new ContactsPresenter(view, api, configuration, usersPersistanceDB);

    }
}

File: fintechplatform-ui/src/main/java/com/fintechplatform/ui/profile/contacts/di/ContactsPresenterModule.kt
package com.fintechplatform.ui.profile.contacts.di

import com.fintechplatform.api.profile.api.ProfileAPI
import com.fintechplatform.ui.models.DataAccount
import com.fintechplatform.ui.profile.contacts.ContactsContract
import com.fintechplatform.ui.profile.contacts.ContactsPresenter
import com.fintechplatform.ui.profile.db.user.UsersPersistanceDB
import dagger.Module
import dagger.Provides
import javax.inject.Singleton

@Module
class ContactsPresenterModule(private val view: ContactsContract.View, private val configuration: DataAccount) {

    @Provides
    @Singleton
    internal fun providesContactsPresenter(api: ProfileAPI, usersPersistanceDB: UsersPersistanceDB): ContactsContract.Presenter {
        return ContactsPresenter(view, api, configuration, usersPersistanceDB)

    }
}
